### PR TITLE
UrQMD bugfix 

### PR DESCRIPTION
**Pull request by kotania**
* UrQMD assumes a nucleon-nucleon frame as default, which is not the same as CM in impy. To obtain correct distributions in impy in the CM frame, one needs to set lab frame as a default in UrQMD source (see  `src/urqmd-3.4/sources/input.f`, **CTOption(27)** ) and boost to CM manually within impy (if needed, otherwise stay in lab). 

* To get correct distributions for heavy nuclei collisions, frozen Fermi motion has to be disabled (i.e. allowing the nucleons to interact before collision). This is done in `src/urqmd-3.4/sources/urqinit.f` , **CTOption(30)**. 

* Other commits include slicing in the decay afterburner (the old `event[append]` does not work if append is outside of the event length range) and fixing the center-of-mass momentum definition in `impy.kinematics` (dimensions were wrong before).

### CODE REVIEW

**Review by afedynitch — COMMENTED** [review]
I left some questions regarding a some of the parameter changes. It would be good to know some background before moving forward.

File: src/urqmd-3.4/sources/urqinit.f [hunk 1]
```diff
@@ -240,43 +240,43 @@ SUBROUTINE URQINI(MDEBUG, iudebug)
...
+      CTOption(4)=0
       CTOStrng(4)=' no initial conf. output '
-      CTOption(5)=0  
+      CTOption(5)=0
       CTOStrng(5)=' fixed impact parameter'
-      CTOption(6)=0  
+      CTOption(6)=0
       CTOStrng(6)=' no first collisions inside proj/target'
-      CTOption(7)=0  
+      CTOption(7)=1
```
[the remark this answers is not in the document]
Comment (kotania) [reply src/urqmd-3.4/sources/urqinit.f:255]: Exactly! 0 = elastic collisions are allowed, 1 = no elastic NN collisions. So I'm proposing to have no elastic collisions by default.

File: src/urqmd-3.4/sources/urqinit.f [hunk 2]
```diff
@@ -347,14 +347,14 @@ SUBROUTINE URQINI(MDEBUG, iudebug)
...
-      CTOStrng(55)=' f19 output adjusted for visualization'  
+      CTOStrng(55)=' f19 output adjusted for visualization'
       CTOPtion(56)=0
       CTOStrng(56)=' f15 output has unique particle id'
       CTOPtion(57)=1
       CTOStrng(57)=' legacy event header w/ missing cto and ctp'
       CTOPtion(58)=0
       CTOStrng(58)=' standard event header in collision file (file15)'
-      CTOption(59)=1
+      CTOption(59)=0
```
[the remark this answers is not in the document]
Comment (kotania) [reply src/urqmd-3.4/sources/urqinit.f:357]: This flag was set to "0" to disable possible strangeness exchanges in reactions involving the Lambda baryon (which would have the potential of explaining the observed "backwards" propagation of Lambda projectiles after collision). However, as this flag had no effect in the end, it is OK to keep 1 as the default.

File: src/urqmd-3.4/sources/urqinit.f [hunk 3]
```diff
@@ -336,7 +336,7 @@ SUBROUTINE URQINI(MDEBUG, iudebug)
       CTOStrng(47)=' flag for equation of state for hydro'
       CTOption(48)=0
       CTOStrng(48)=' propagate only N timesteps of hydro evolution'
-      CTOption(49)=0
+      CTOption(49)=1
```
[the remark this answers is not in the document]
Comment (kotania) [reply src/urqmd-3.4/sources/urqinit.f:339]: Like CTOption(59), this flag was reset from its default value (spectator propagation in UrQMD) to "spectator propagation as part of the fluid" as a safety measure in the Lambda-nucleon collision experiments, where non-interacting Lambda-nucleon pairs seem to swap labels upon collision. The idea was to investigate whether UrQMD vs. "fluid" propagation do anything different in terms of spectator label assignment. The outcome is that they don't, which is why it is also OK to switch this option back to default.

File: src/urqmd-3.4/sources/urqinit.f (hunk 2 above)
[the remark this answers is not in the document]
Comment (afedynitch) [reply src/urqmd-3.4/sources/urqinit.f:357]: ok, thanks

File: src/urqmd-3.4/sources/urqinit.f (hunk 1 above)
[the remark this answers is not in the document]
Comment (afedynitch) [reply src/urqmd-3.4/sources/urqinit.f:255]: Cool. This simplifies some issues! Always misused the flag...

**Comment by afedynitch** [comment]
Thanks so much for finding and correcting the issue! Your PR is now merged

**Pull request merged**